feat: Rebranding (renaming) `fury-distribution` into `sighup-distribution`

nutellinoit opened the pull request:
### Pre-submission checklist:

*Please check each of these after submitting your pull request:*

* [x] If this is a new entry, have you submitted a signed [participation form](https://github.com/cncf/k8s-conformance/tree/master/participation-form)?
* [x] Did you include the product/project logo in SVG, EPS or AI format?  
* [x] Does your logo clearly state the name of the product/project and follow the other logo [guidelines](https://github.com/cncf/landscape#logos)?
* [x] If your product/project is open source, did you include the `repo_url`?
* [x] Did you copy and paste the [installation and configuration instructions](https://github.com/cncf/k8s-conformance/blob/master/faq.md#can-i-provide-a-link-to-the-installation-directions) into the README.md file in addition to linking to them?

For a full list of requirements, please refer to these sections of the docs: [_Contents of the PR_](https://github.com/cncf/k8s-conformance/blob/master/instructions.md#contents-of-the-pr), and [_Requirements_](https://github.com/cncf/k8s-conformance/blob/master/instructions.md#requirements).

### Description

This PR renames fury-distribution into sighup-distribution since SIGHUP has been acquired by ReeVo. I also submitted by DocuSign the updated participation form, I hope it's ok.

taylorwaggoner commented:
@nutellinoit thanks for submitting. A couple things here: 
- Please update the PRODUCT file to link to a product logo in SVG, AI, or EPS format. We cannot accept PNGs. 
- We only need to update the most recent results for your product with the new name. Could you please keep 1.31 and remove the other versions? 

Thanks!

nutellinoit commented:
> Sorry for the confusion, I just meant you do not need to rename them in this pull request. They can remain in the repository under the previous name. Thanks!

Hi @taylorwaggoner , I did the requested changes, I updated only the last version and replaced the logo with an SVG logo

nutellinoit commented:
> @nutellinoit thanks! Could you please update the product logo to one that includes either the product name or company name? Thanks!

Hi @taylorwaggoner , I replaced the logo with one with the product name, I think it should be ok now

taylorwaggoner commented:
Thanks! I've submitted the requested changes on the Cloud Native Landscape, so you're good to go.
